contracts/anchor-handler/src/contract.rs

```diff
@@ -0,0 +1,371 @@
+        // by a relayer on the deposit's destination chain.
+        //
+        // Params:
+        //   "resource_id": ResourceID corresponding to a particular set of Anchors
+        //   "data": Consists of {sourceChainID}, {leafIndex}, {merkleRoot} all padded to 32 bytes.
+        //
+        // Data passed into the function should be constructed as follows:
+        // chainID                                  u64          bytes  0 - 32
+        // leafIndex                                u64          bytes  32 - 64
+        // merkleRoot                               [u8; 32]     bytes  64 - 96
+        ExecuteMsg::ExecuteProposal { resource_id, data } => {
+            execute_proposal(deps, info, resource_id, data)
+        } /* ---------------------------------------- */
+    }
+}
+
+fn exec_set_resource(
+    deps: DepsMut,
+    info: MessageInfo,
+    resource_id: [u8; 32],
+    contract_addr: String,
+) -> Result<Response, ContractError> {
+    let state = STATE.load(deps.storage)?;
+
+    // Validations
+    if info.sender != state.bridge_addr {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Save/update the mapping `resource_id => contract_addr`
+    let contract_addr = deps.api.addr_validate(&contract_addr)?;
+    set_resource(deps.storage, resource_id, contract_addr)?;
+
+    Ok(Response::new().add_attribute("method", "set_resource"))
+}
+
+fn migrate_bridge(
+    deps: DepsMut,
+    info: MessageInfo,
+    new_bridge: String,
+) -> Result<Response, ContractError> {
+    let state = STATE.load(deps.storage)?;
+
+    // Validations
+    if info.sender != state.bridge_addr {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Migrage(update) the "bridge_addr" with "new_bridge"
+    let bridge_addr = deps.api.addr_validate(&new_bridge)?;
+    STATE.save(deps.storage, &State { bridge_addr })?;
+
+    Ok(Response::new().add_attribute("method", "migrate_bridge"))
+}
+
+fn execute_proposal(
+    deps: DepsMut,
+    info: MessageInfo,
+    resource_id: [u8; 32],
+    data: Vec<u8>,
+) -> Result<Response, ContractError> {
+    // Parse the `data`.
+    let resource_ID = element_encoder(&data);
+    let func_sig = bytes4_encoder(&data[32..36]);
+    let arguments = &data[36..];
+
+    let state = STATE.load(deps.storage)?;
+
+    // Validations
+    if info.sender != state.bridge_addr {
+        return Err(ContractError::Unauthorized {});
+    }
+    let anchor_addr = read_contract_addr(deps.storage, resource_id)?;
+    if !read_whitelist(deps.storage, anchor_addr)? {
+        return Err(ContractError::Std(StdError::GenericErr {
+            msg: "provided tokenAddress is not whitelisted".to_string(),
+        }));
+    }
+
+    // Execute the proposal according to function signature
+    let set_handler_sig = bytes4_encoder(
+        &Keccak256::hash(b"setHandler(address,uint32)").map_err(|_| ContractError::DecodeError)?,
```

Comment:
Can we "mock" the function signatures as they are on CosmWasm? For example `uint32` would be `u32`, etc.